Storage/container/one2one.h

```diff
@@ -0,0 +1,253 @@
+    } else {
+      const auto it_row = forward_.find(row);
+      const auto it_col = transposed_.find(col);
+      bool row_occupied = it_row != forward_.end();
+      bool col_occupied = it_col != transposed_.end();
+      if (row_occupied && col_occupied) {
+        const T prev_object1 = *(it_row->second);
+        const T prev_object2 = *(it_col->second);
+        const auto row_col1 = std::make_pair(row, sfinae::GetCol(prev_object1));
+        const auto row_col2 = std::make_pair(sfinae::GetRow(prev_object2), col);
+        journal_.LogMutation(T_DELETE_EVENT(prev_object1),
+                             [this, row_col1, prev_object1]() { DoAdd(row_col1, prev_object1); });
+        journal_.LogMutation(T_DELETE_EVENT(prev_object2),
+                             [this, row_col2, prev_object2]() { DoAdd(row_col2, prev_object2); });
+        DoErase(row_col1);
+        DoErase(row_col2);
+      } else if (row_occupied || col_occupied) {
+        const T prev_object = row_occupied ? *(it_row->second) : *(it_col->second);
+        const auto prev_row_col = std::make_pair(sfinae::GetRow(prev_object), sfinae::GetCol(prev_object));
+        journal_.LogMutation(T_DELETE_EVENT(prev_object),
+                             [this, prev_row_col, prev_object]() { DoAdd(prev_row_col, prev_object); });
+        DoErase(prev_row_col);
+      }
+    }
+    journal_.LogMutation(T_UPDATE_EVENT(object), [this, row_col]() { DoErase(row_col); });
+    DoAdd(row_col, object);
+  }
+
+  void Erase(const T_RCPAIR& row_col) {
+    auto it = map_.find(row_col);
+    if (it != map_.end()) {
+      const T prev_object = *(it->second);
+      journal_.LogMutation(T_DELETE_EVENT(prev_object),
+                           [this, row_col, prev_object]() { DoAdd(row_col, prev_object); });
+      DoErase(row_col);
+    }
+  }
+  void Erase(sfinae::CF<T_ROW> row, sfinae::CF<T_COL> col) { Erase(std::make_pair(row, col)); }
+
+  void EraseRow(sfinae::CF<T_ROW> row) {
+    const auto it = forward_.find(row);
+    if (it != forward_.end()) {
+      const T prev_object = *(it->second);
+      const auto row_col = std::make_pair(row, sfinae::GetCol(prev_object));
+      journal_.LogMutation(T_DELETE_EVENT(prev_object),
+                           [this, row_col, prev_object]() { DoAdd(row_col, prev_object); });
+      DoErase(row_col);
+    }
+  }
+
+  void EraseCol(sfinae::CF<T_COL> col) {
+    const auto it = transposed_.find(col);
+    if (it != transposed_.end()) {
+      const T prev_object = *(it->second);
+      const auto row_col = std::make_pair(sfinae::GetRow(prev_object), col);
+      journal_.LogMutation(T_DELETE_EVENT(prev_object),
+                           [this, row_col, prev_object]() { DoAdd(row_col, prev_object); });
+      DoErase(row_col);
+    }
+  }
+
+  ImmutableOptional<T> operator[](const T_RCPAIR& row_col) const {
+    const auto it = map_.find(row_col);
+    if (it == map_.end()) return nullptr;
```

Comment:
Yes please. It's from [Google Style Guide](http://google.github.io/styleguide/cppguide.html).

"... some projects require that an if must always always have an accompanying brace."

```
if (condition)
  DoSomething();  // 2 space indent.

if (condition) {
  DoSomething();  // 2 space indent.
}
```

(ignore the "some projects" part, it's only because they merged a couple loosely related C++ guides together.)